connect-inject/health_check_resource_test.go

```diff
@@ -0,0 +1,553 @@
+package connectinject
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/consul-k8s/helper/controller"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	testPodName               = "test-pod"
+	testServiceNameAnnotation = "test-service"
+	testServiceNameReg        = "test-pod-test-service"
+	testHealthCheckID         = "default_test-pod-test-service_kubernetes-health-check-ttl"
+)
+
+func testServerAgentResourceAndController(t *testing.T, pod *corev1.Pod) (*testutil.TestServer, *api.Client, *HealthCheckResource, *controller.Controller) {
+	require := require.New(t)
+	// Set up server, client
+	s, err := testutil.NewTestServerConfigT(t, nil)
+	require.NoError(err)
+
+	clientConfig := &api.Config{Address: s.HTTPAddr}
+	require.NoError(err)
+	client, err := api.NewClient(clientConfig)
+	require.NoError(err)
+
+	healthResource := HealthCheckResource{
+		Log:          hclog.Default().Named("healthCheckResource"),
+		Clientset:    fake.NewSimpleClientset(pod),
+		ClientConfig: api.DefaultConfig(),
+		ConsulPort:   strings.Split(s.HTTPAddr, ":")[1],
+		SyncPeriod:   0,
+	}
+
+	ctl := &controller.Controller{
+		Log:      hclog.Default().Named("healthCheckController"),
+		Resource: &healthResource,
+	}
```

Comment:
I don't think either of the tests is calling the controller except for the `registerHealthCheck` function below. In this function, the controller is also not strictly needed since it's used only for logging.